specification/nginx/resource-manager/NGINX.NGINXPLUS/preview/2024-01-01-preview/swagger.json

```diff
@@ -0,0 +1,1657 @@
+    "NginxPrivateIPAllocationMethod": {
+      "enum": [
+        "Static",
+        "Dynamic"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "NginxPrivateIPAllocationMethod"
+      }
+    },
+    "NginxPrivateIPAddress": {
+      "type": "object",
+      "properties": {
+        "privateIPAddress": {
+          "type": "string"
+        },
+        "privateIPAllocationMethod": {
+          "$ref": "#/definitions/NginxPrivateIPAllocationMethod"
+        },
+        "subnetId": {
+          "type": "string"
+        }
+      }
+    },
+    "NginxPublicIPAddress": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string"
+        }
+      }
+    },
+    "NginxFrontendIPConfiguration": {
+      "type": "object",
+      "properties": {
+        "publicIPAddresses": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NginxPublicIPAddress"
+          },
+          "x-ms-identifiers": []
+        },
+        "privateIPAddresses": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NginxPrivateIPAddress"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "NginxNetworkInterfaceConfiguration": {
+      "type": "object",
+      "properties": {
+        "subnetId": {
+          "type": "string"
+        }
+      }
+    },
+    "NginxNetworkProfile": {
+      "type": "object",
+      "properties": {
+        "frontEndIPConfiguration": {
+          "$ref": "#/definitions/NginxFrontendIPConfiguration"
+        },
+        "networkInterfaceConfiguration": {
+          "$ref": "#/definitions/NginxNetworkInterfaceConfiguration"
+        }
+      }
+    },
+    "NginxStorageAccount": {
+      "type": "object",
+      "properties": {
+        "accountName": {
+          "type": "string"
+        },
+        "containerName": {
+          "type": "string"
+        }
+      }
+    },
+    "NginxLogging": {
+      "type": "object",
+      "properties": {
+        "storageAccount": {
+          "$ref": "#/definitions/NginxStorageAccount"
+        }
+      }
+    },
+    "NginxDeploymentScalingProperties": {
+      "type": "object",
+      "description": "Information on how the deployment will be scaled.",
+      "properties": {
+        "capacity": {
+          "format": "int32",
+          "type": "integer",
+          "example": 20
+        },
+        "autoScaleSettings": {
+          "type": "object",
+          "required": [
+            "profiles"
+          ],
+          "description": "The settings for enabling automatic scaling of the deployment. If this field is specified, 'scale.capacity' must be empty.",
+          "x-ms-client-flatten": true,
+          "properties": {
+            "profiles": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/ScaleProfile"
+              },
+              "x-ms-identifiers": []
```

Comment:
>"x-ms-identifiers": [] [](http://example.com/codeflow?start=14&length=22)

Looks like the name (or scaleProfileName) property should go here. Will it be unique in this collection?

Reply:
Hi @mentat9 

I am curious to know what you mean by this comment. The `ScaleProfile` is an array and each element/profile gets a name.